packages/components/pie-checkbox/src/index.ts

```diff
@@ -36,7 +35,10 @@ export class PieCheckbox extends FormControlMixin(RtlMixin(LitElement)) implemen
     public name?: CheckboxProps['name'];
 
     @property({ type: Boolean, reflect: true })
-    public checked = defaultProps.checked;
```

Comment:
Question so I can understand better: I think I passed it this way because of the lint error "`boolean | undefined` is not assignable to boolean." you mentioned below. You fixed it by bringing it to boolean via `!!`.  I guess we never should use direct referencing to defaultProps like I did here, right?

Reply:
@dandel10n It's a good point. I think the issue is that we insist on making our properties optional in the TypeScript types, so if we pass that through into the component we are forced to resolve the possible undefined (even though we know it will always have a value because we assign a default).

I think there are two ways to deal with this:
1. Don't use `CheckboxProps` in this file to force `boolean` props to `boolean | undefined`; as long as we provide a default value it should never be undefined inside the component.
2. Continue to use `CheckboxProps`, but use `Boolean()` or `!!` to make TypeScript see that it will always have a value.

I think I would prefer option number 1 because it's a bit cleaner. We also don't lose JSDoc comments as those are still picked up because the class `implements CheckboxProps`.

As you're away this week, I'll revert it and we can discuss aligning the rest of the props when you're back.